bug: requeuePendingMsgs Deadlock and Local Dispatch Contention

## CI Failure

```text
panic: test timed out after 10m0s
    running tests:
        TestQuorumCallPartialFailures (10m0s)
        TestQuorumCallPartialFailures/QuorumCall/First/fail=3 (9m59s)
```

### Root Cause

`requeuePendingMsgs` is called from `sender()` after a `stream.Send` failure. It calls `c.Enqueue(req)`, which tries `c.sendQ <- req`. With the default `sendBuffer: 0` (unbuffered sendQ), this write blocks until someone reads. **The sender goroutine is the sole reader of sendQ — and it's the one calling Enqueue.** We have a self-deadlock.

### CI Stack Trace Evidence

Two goroutines stuck on the same channel (`0x8d5aff9bcc0`), both at `channel.go:288`:

**Goroutine 3554** (sender goroutine — deadlocked):

```text
Channel.Enqueue              channel.go:288   ← c.sendQ <- req BLOCKS
Channel.requeuePendingMsgs   channel.go:361
Channel.sender               channel.go:454
```

**Goroutine 3323** (test goroutine — blocked because sender can't read):

```text
Channel.Enqueue              channel.go:288   ← c.sendQ <- req BLOCKS
Node.Enqueue                 node.go:191
ClientCtx.send               client_interceptor.go:204
sync.Once.Do                 (sendOnce.Do)
defaultResponseSeq           client_interceptor.go:240
Responses.Threshold          responses.go:221
Responses.First              responses.go:198
TestQuorumCallPartialFailures quorumcall_test.go:119
```
The sender is stuck writing to its own unbuffered sendQ (nobody reads). The test's `send()` is also stuck trying to enqueue to the same channel. Result: permanent hang until the 10-minute test timeout kills the process.

```text
2026-03-14T13:36:04.2681355Z panic: test timed out after 10m0s
2026-03-14T13:36:04.2681888Z 	running tests:
2026-03-14T13:36:04.2682355Z 		TestQuorumCallPartialFailures (10m0s)
2026-03-14T13:36:04.2683192Z 		TestQuorumCallPartialFailures/QuorumCall/First/fail=3 (9m59s)
2026-03-14T13:36:04.2683726Z 
[...]
2026-03-14T13:36:04.2714568Z goroutine 3323 [select, 9 minutes]:
2026-03-14T13:36:04.2715236Z github.com/relab/gorums/internal/stream.(*Channel).Enqueue(0x8d5aff9bcc0, {{0xb55a78, 0x8d5aaa51320}, 0x8d5a6c70bd0, 0x0, 0x0, 0x8d5a6c70b60, {0x0, 0x0, 0x0}})
2026-03-14T13:36:04.2716009Z 	/home/runner/work/gorums/gorums/internal/stream/channel.go:288 +0x1b0
2026-03-14T13:36:04.2716527Z github.com/relab/gorums.(*Node).Enqueue(...)
2026-03-14T13:36:04.2717185Z 	/home/runner/work/gorums/gorums/node.go:191
2026-03-14T13:36:04.2717737Z github.com/relab/gorums.(*ClientCtx[...]).send(0xb64200)
2026-03-14T13:36:04.2718465Z 	/home/runner/work/gorums/gorums/client_interceptor.go:204 +0x24f
2026-03-14T13:36:04.2719329Z sync.(*Once).doSlow(0x8d5b061bcc8?, 0x7f8fb05801c8?)
2026-03-14T13:36:04.2720003Z 	/opt/hostedtoolcache/go/1.26.0/x64/src/sync/once.go:78 +0xac
2026-03-14T13:36:04.2720602Z sync.(*Once).Do(0x427234?, 0x7f8ff9271108?)
2026-03-14T13:36:04.2721162Z 	/opt/hostedtoolcache/go/1.26.0/x64/src/sync/once.go:69 +0x19
2026-03-14T13:36:04.2721616Z github.com/relab/gorums.(*clientCtxBuilder[...]).defaultResponseSeq.func2()
2026-03-14T13:36:04.2722351Z 	/home/runner/work/gorums/gorums/client_interceptor.go:240 +0x65
2026-03-14T13:36:04.2722789Z github.com/relab/gorums.(*Responses[...]).Threshold(0xb64320, 0x1)
2026-03-14T13:36:04.2723313Z 	/home/runner/work/gorums/gorums/responses.go:221 +0x159
2026-03-14T13:36:04.2723925Z github.com/relab/gorums.(*Responses[...]).First(...)
2026-03-14T13:36:04.2724512Z 	/home/runner/work/gorums/gorums/responses.go:198
2026-03-14T13:36:04.2725723Z github.com/relab/gorums_test.TestQuorumCallPartialFailures.TestQuorumCallPartialFailures.func1.func19(0x2faf080?, 0x8d5a5ea4d88?)
2026-03-14T13:36:04.2727036Z 	/home/runner/work/gorums/gorums/quorumcall_test.go:119 +0x49
2026-03-14T13:36:04.2727806Z github.com/relab/gorums_test.TestQuorumCallPartialFailures.func3(0x8d5a5ea4d88)
2026-03-14T13:36:04.2728497Z 	/home/runner/work/gorums/gorums/quorumcall_test.go:190 +0x24d
2026-03-14T13:36:04.2728884Z testing.tRunner(0x8d5a5ea4d88, 0x8d5ab2fb940)
2026-03-14T13:36:04.2729752Z 	/opt/hostedtoolcache/go/1.26.0/x64/src/testing/testing.go:2036 +0xea
2026-03-14T13:36:04.2730404Z created by testing.(*T).Run in goroutine 2129
2026-03-14T13:36:04.2731086Z 	/opt/hostedtoolcache/go/1.26.0/x64/src/testing/testing.go:2101 +0x4c5
2026-03-14T13:36:04.2731554Z 
[...]
2026-03-14T13:36:04.2807056Z goroutine 3554 [select, 9 minutes]:
2026-03-14T13:36:04.2808375Z github.com/relab/gorums/internal/stream.(*Channel).Enqueue(0x8d5aff9bcc0, {{0xb55a78, 0x8d5aaa51320}, 0x8d5a5f35810, 0x0, 0x0, 0x8d5a5f35730, {0xc265765b5f61630a, 0x4bbd3e18, 0x10b4820}})
2026-03-14T13:36:04.2810067Z 	/home/runner/work/gorums/gorums/internal/stream/channel.go:288 +0x1b0
2026-03-14T13:36:04.2810628Z github.com/relab/gorums/internal/stream.(*Channel).requeuePendingMsgs(0x8d5aff9bcc0)
2026-03-14T13:36:04.2811466Z 	/home/runner/work/gorums/gorums/internal/stream/channel.go:361 +0xb6
2026-03-14T13:36:04.2812271Z github.com/relab/gorums/internal/stream.(*Channel).sender(0x8d5aff9bcc0)
2026-03-14T13:36:04.2813118Z 	/home/runner/work/gorums/gorums/internal/stream/channel.go:454 +0x4e5
2026-03-14T13:36:04.2813971Z created by github.com/relab/gorums/internal/stream.newChannel in goroutine 3323
2026-03-14T13:36:04.2814866Z 	/home/runner/work/gorums/gorums/internal/stream/channel.go:129 +0x2c5
```

### Trigger Scenario

`TestQuorumCallPartialFailures/QuorumCall/First/fail=3` stops all 3 servers, then calls `First` which attempts to send to 3 nodes. The stopped servers cause `stream.Send` failures. The sender calls `requeuePendingMsgs` → `Enqueue` → self-deadlock.

--- a/internal/stream/channel.go
+++ b/internal/stream/channel.go
@@ -357,11 +357,24 @@ func (c *Channel) cancelPendingMsgs(err error) {
 // with ErrStreamDown because they cannot be safely retried.
 func (c *Channel) requeuePendingMsgs() {
 	requeue, cancel := c.router.RequeuePending()
-	for _, req := range requeue {
-		c.Enqueue(req)
-	}
 	for _, req := range cancel {
 		c.replyError(req, ErrStreamDown)
+	}
+	// The requeue is performed in a separate goroutine because this method is called
+	// from sender(), which is the sole reader of sendQ. Calling Enqueue directly from
+	// the sender would deadlock: Enqueue writes to the unbuffered sendQ, but no one
+	// is reading it because the sender itself is blocked in the Enqueue call.
+	// The goroutine writes to sendQ while the sender returns to its read loop.
+	//
+	// If connCtx is cancelled before the goroutine finishes, each Enqueue call will
+	// take the connCtx.Done() branch and replyError with ErrNodeClosed, and
+	// drainSendQ (deferred in sender) will drain any items that made it to sendQ.
+	if len(requeue) > 0 {
+		go func() {
+			for _, req := range requeue {
+				c.Enqueue(req)
+			}
+		}()
 	}
 }
 

